Assets/Scripts/MusicTransitionTrigger.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MusicTransitionTrigger : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The music to trigger when the player enters.")]
+    private MusicManager.CurrentMusic musicToTrigger;
+
+    [SerializeField]
+    [Tooltip("The MusicManager in the scene.")]
+    private MusicManager musicManager;
```

Comment:
you could probably simplify this by using `FindObjectOfType` to initialize rather than dragging it in, since there's only one. I think this is better because you can't forget to drag it in.